### PR TITLE
Use `Sentry.framework` instead of `Sentry.xcframework`

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Replaced `Sentry.xcframework` with `Sentry.framework` in the watchOS sample, so it is in line with the other apps.

## :bulb: Motivation and Context

Relates to [#4300](https://github.com/getsentry/sentry-cocoa/issues/4300)

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by brustolin** [comment]
> The watchOS sample was not building

I think this is by design, we have a script that make it work. I dont know the reason tho.
If nothing is breaking I guess we could change that.
@philipphofmann is back in a week if I'm not mistaken, we can double check with him.

**Comment by armcknight** [comment]
Yeah I'm not sure why the projects weren't set up the same way. I am pushing a small commit to try to fix the build > watchOS sample CI failure. I made the watchOS app's project and linkages mirror iOS-Swift's. 

Before (on this branch as of my review):

image

After my commit:

image

**Comment by armcknight** [comment]
So one more thing I saw when trying to reproduce that last failure locally, is that if you build the watchOS App scheme, it automatically builds the sentry sdk for iphone, and then can't find the build product because it's not in the right derived data tree. But if you build the _extension_, it gets it right. Pushing a commit to change that.

image

File: Sources/Configuration/SDK.xcconfig [hunk 1]
```diff
@@ -17,7 +17,6 @@ CLANG_ENABLE_OBJC_ARC = YES
 CLANG_ENABLE_MODULES = YES
 //OTHER_CFLAGS = -Wall -Wextra -Wpedantic -Wno-gnu-conditional-omitted-operand
 
-ONLY_ACTIVE_ARCH[config=Debug] = YES
```
Comment (armcknight) [inline Sources/Configuration/SDK.xcconfig:20]: This did nothing so I just removed it as part of cleaning things up
image

**Comment by armcknight** [comment]
I'm having trouble finding the right invocation for the watchOS sample app build to succeed in CI 😕

**Comment by denrase** [comment]
@armcknight Thx for the info, i will check if i can make it run on ci!

**Comment by denrase** [comment]
@armcknight Doing the same in CI as for the macOS and iOS seems to work.

@philipphofmann I removed the code to generate the xcframework just for the watchOS sample project.

**Comment by denrase** [comment]
@armcknight Do we still need the `watchOS-Swift WatchKit Extension.xcscheme` or can it be removed again?

**Comment by armcknight** [comment]
@denrase I'm not 100% sure. But if you got it working without it, then chuck it! We can always add it back and figure it out if we need it again for some reason.

**Review by armcknight — APPROVED** [review]
Thanks @denrase , so much nicer to have this work consistently like the other sample apps.

**Pull request merged**